packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,155 @@
+/// tree to remain marked not sensitive as long as there are (1) no other `SensitiveContent`
+/// widgets in the tree or (2) there  are only other not sensitive `SensitiveContent` widgets
+/// in the tree. If there are no `SensitiveContent` widgets in the tree, the default setting as
+/// queried from the embedding will be used. This could be set by a Flutter developer in native
+/// Android; otherwise, Android uses [ContentSensitivity.autoSensitive] by default; see
+/// https://developer.android.com/reference/android/view/View#getContentSensitivity().
+/// {@endtemplate}
+///
+/// * See `SensitiveContent` for how to set a [ContentSensitivity] level
+///   in order for sensitive content to be obscured when the Flutter screen
+///   is shared.
+enum ContentSensitivity {
+  /// Content sensitivity is auto-detected by the native platform.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will only be marked as sensitive if other `SensitiveContent`
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_AUTO for how
+  /// this mode behaves on native Android.
+  ///
+  /// For Android `View`s, this mode attempts to auto detect passwords, 2factor tokens, and other
+  /// sensitive content. As of API 35, Android cannot determine if Flutter `View`s contain sensitive
+  /// content automatically, and thus will never obscure the screen.
+  // TODO(camsim99): Implement `autoSensitive` mode that matches the behavior
+  // of `CONTENT_SENSITIVITY_AUTO` on Android that has implemented based on autofill hints; see
+  // https://github.com/flutter/flutter/issues/160879.
+  autoSensitive,
+
+  /// The widget tree contains sensitive content.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will be marked as sensitive during an active media
+  /// projection session.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_SENSITIVE.
+  sensitive,
+
+  /// The widget tree does not contain sensitive content.
+  ///
+  /// When this level is set via a `SensitiveContent` widget, the window
+  /// hosting the screen will only be marked as sensitive if other `SensitiveContent`
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_NOT_SENSITIVE.
+  notSensitive,
+
+  /// The sensitivity content level is unknown to Flutter.
+  ///
+  /// This mode may represent the current content sensitivity of the window if, for example, Android
+  /// adds a new mode that is not recognized by the `SensitiveContent` widget.
+  ///
+  /// This mode cannot be used to set the sensitivity level of a `SensitiveContent` widget.
+  _unknown,
```

Comment:
for cases where you send _unknown through channel, can we just return error instead?

Reply:
@reidbaker and I discussed this at length. I'm sure we could accomplish the same thing that way, as I mentioned before, but this gives a standardized way for any platform to signal that an unknown value was detected and we should gracefully fail and not crash the app.

Reply:
sgtm